packages/amplify-graphql-auth-transformer/src/vtl-generator/rds/resolvers/common.ts

```diff
@@ -211,10 +212,19 @@ export const constructAuthorizedInputStatement = (keyName: string): Expression =
 /**
  * Generates sandbox expression for field
  */
-export const generateSandboxExpressionForField = (sandboxEnabled: boolean): string => {
-  let exp: Expression;
-  if (sandboxEnabled) exp = iff(notEquals(methodCall(ref('util.authType')), str(API_KEY_AUTH_TYPE)), methodCall(ref('util.unauthorized')));
-  else exp = ref('util.unauthorized()');
+export const generateSandboxExpressionForField = (sandboxEnabled: boolean, genericIamAccessEnabled: boolean): string => {
+  let exp: Expression = methodCall(ref('util.unauthorized'));
+  if (sandboxEnabled) {
+    exp = iff(notEquals(methodCall(ref('util.authType')), str(API_KEY_AUTH_TYPE)), exp);
+  }
+  if (genericIamAccessEnabled) {
+    const isNonCognitoIAMPrincipal = and([
+      equals(ref('util.authType()'), str(IAM_AUTH_TYPE)),
+      methodCall(ref('util.isNull'), ref('ctx.identity.cognitoIdentityPoolId')),
+      methodCall(ref('util.isNull'), ref('ctx.identity.cognitoIdentityId')),
+    ]);
```

Comment:
I see this block is used atleast in two places. Can we move it to a helper method?

Reply:
I moved things around in https://github.com/aws-amplify/amplify-category-api/pull/2331/commits/b70d224ab3ce645a0ac1e3598611efd5ee7fcd8e .

Notice that:
1. I didn't change model transformer as it doesn't share code with auth transformer (I don't think it's worth moving this piece to transformer core, there's already precedent of auth stuff being copy-simplify-paste to model transformer)
2. I also moved other expression that is now shared between rds and ddb.